src/Infrastructure/Payment/PayPalPaymentNotificationVerifier.php

```diff
@@ -69,19 +82,21 @@ public function verify( array $request ): void {
 		}
 
 		$responseBody = trim( $result->getBody()->getContents() );
+		if ( $responseBody === 'VERIFIED' ) {
+			return;
+		}
+
```

Comment:
You could set the error message string here based on the $responseBody and only define the PayPalPaymentNotificationVerifierException throw once.